spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/catalyst/analysis/RewriteArcticCommand.scala

```diff
@@ -39,6 +43,35 @@ case class RewriteArcticCommand(sparkSession: SparkSession) extends Rule[Logical
       case t@TruncateTable(r: ResolvedTable, partitionSpec)
         if isArcticTable(r.table) =>
         TruncateArcticTable(t.child, partitionSpec)
+      case c@CreateTableAsSelect(catalog, _, _, _, props, options, _) if isArcticCatalog(catalog) =>
+        var propertiesMap: Map[String, String] = props
+        var optionsMap: Map[String, String] = options
+        if (options.contains("primary.keys")) {
+          propertiesMap += ("primary.keys" -> options("primary.keys"))
+        }
+        if (propertiesMap.contains("primary.keys")) {
+          optionsMap += (WriteMode.WRITE_MODE_KEY -> WriteMode.OVERWRITE_DYNAMIC.mode)
+        }
+        c.copy(properties = propertiesMap, writeOptions = optionsMap)
+      case CreateTableLikeCommand(targetTable, sourceTable, storage, provider, properties, ifNotExists)
+        if provider.get != null && provider.get.equals("arctic") =>
+          val sourceIdentifier = buildArcticIdentifier(sparkSession, sourceTable)
+          val targetIdentifier = buildArcticIdentifier(sparkSession, targetTable)
+          val arcticCatalog = buildCatalog(sourceIdentifier)
+          val table = arcticCatalog.loadTable(sourceIdentifier.identifier())
```

Comment:
arcticCatalog => sourceCatalog  maybe better? create table like command source table could be a HiveTable or V1DataSourceTable or some other V2Table